### DISCUSSION TITLE
how does rc4_hash__  calculate?

### DISCUSSION THREAD

**Discussion by fswen16** [PiotrMachowski/Xiaomi-cloud-tokens-extractor#116] [Q&A]
Hello, I utilized this code to decrypt my Mi Band 7 and successfully got the data, but I've encountered an issue: I employed 'generate_enc_params' and 'generate_enc_signature' as separate functions. Then, using the same 'ssecurity', 'nonce', and 'params' from the captured packet, I attempted to generate the 'rc4_hash__'. However, I can't seem to get the same 'rc4_hash__' of the packet I captured. Could there be an error in my 'signature_string'? Your assistance would be greatly appreciated.

**Comment by PiotrMachowski** [comment]
It's hard to tell, it might be a different order of params or a different formatting of values in params
